### PR TITLE
[GFX11] Batchnorm new target changes

### PR DESCRIPTION
**Pull request by muralinr**
Batchnorm new target changes

### CODE REVIEW

File: src/kernels/MIOpenBatchNormActivBwdPerAct.cl [hunk 1]
```diff
@@ -39,6 +39,12 @@
 #define MIOPEN_USE_AMDGCN 1
 #endif
 
+#define MIOPEN_USE_AMDGCN 0
+#if defined(__AMDGCN__) && !MIO_BN_GFX110X
+#undef MIOPEN_USE_AMDGCN
+#define MIOPEN_USE_AMDGCN 1
+#endif
+
```
Comment (atamazov) [inline src/kernels/MIOpenBatchNormActivBwdPerAct.cl]: AFAICS here is a logic error that causes `MIOPEN_USE_AMDGCN` to ~always be 1~ be 1 for gfx10. Don't add these lines. Instead, replace line 37 with
```C++
#if defined(__AMDGCN__) && !(MIO_BN_GFX103X || MIO_BN_GFX110X)
```

File: src/kernels/MIOpenBatchNormActivBwdSpatial.cl [hunk 2]
```diff
@@ -37,6 +37,12 @@
 #define MIOPEN_USE_AMDGCN 1
 #endif
 
+#define MIOPEN_USE_AMDGCN 0
```
Comment (atamazov) [inline src/kernels/MIOpenBatchNormActivBwdSpatial.cl]: Ditto

File: src/kernels/MIOpenBatchNormActivFwdTrainSpatial.cl [hunk 3]
```diff
@@ -38,6 +38,12 @@
 #define MIOPEN_USE_AMDGCN 1
 #endif
 
+#define MIOPEN_USE_AMDGCN 0
```
Comment (atamazov) [inline src/kernels/MIOpenBatchNormActivFwdTrainSpatial.cl]: Ditto

File: src/kernels/MIOpenBatchNormBwdSpatial.cl [hunk 4]
```diff
@@ -38,6 +38,12 @@
 #define MIOPEN_USE_AMDGCN 1
 #endif
 
+#define MIOPEN_USE_AMDGCN 0
```
Comment (atamazov) [inline src/kernels/MIOpenBatchNormBwdSpatial.cl]: Ditto

File: src/kernels/MIOpenBatchNormFwdTrainSpatial.cl [hunk 5]
```diff
@@ -38,6 +38,12 @@
 #define MIOPEN_USE_AMDGCN 1
 #endif
 
+#define MIOPEN_USE_AMDGCN 0
```
Comment (atamazov) [inline src/kernels/MIOpenBatchNormFwdTrainSpatial.cl]: Ditto

File: src/kernels/batchnorm_functions.h [hunk 6]
```diff
@@ -138,6 +138,15 @@
...
 #endif
+
+#ifndef MIOPEN_USE_AMDGCN
+#if defined(__AMDGCN__) && !(defined(MIO_BN_GFX110X) && MIO_BN_GFX110X)
+#define MIOPEN_USE_AMDGCN 1
+#else
+#define MIOPEN_USE_AMDGCN 0
+#endif
+#endif
+
```
Comment (atamazov) [inline src/kernels/batchnorm_functions.h]: Ditto. Please revert this and replace line 135 with
```C++
#if defined(__AMDGCN__) && !( (defined(MIO_BN_GFX103X) && MIO_BN_GFX103X) || (defined(MIO_BN_GFX110X) && MIO_BN_GFX110X) )
```
```

File: src/kernels/batchnorm_functions.h [hunk 7]
```diff
@@ -159,6 +168,10 @@
 #define MIO_BN_GFX103X 0
 #endif
 
+#ifndef MIO_BN_GFX110X
+#define MIO_BN_GFX110X 0
+#endif
+
```
Comment (atamazov) [inline src/kernels/batchnorm_functions.h:167]: [Recommendation] Move this + lines 167-169 somewhere above lines 134-138 and then remove `defined(MIO_BN_GFX103X)` and `defined(MIO_BN_GFX110X)` from line 135. Up to you.

**Review by atamazov — CHANGES_REQUESTED** [review]
Logic problem + some recommendation

File: src/kernels/MIOpenBatchNormActivBwdPerAct.cl (hunk 1 above)
Comment (muralinr) [reply src/kernels/MIOpenBatchNormActivBwdPerAct.cl]: Done!

File: src/kernels/MIOpenBatchNormActivBwdSpatial.cl (hunk 2 above)
Comment (muralinr) [reply src/kernels/MIOpenBatchNormActivBwdSpatial.cl]: Done!

File: src/kernels/MIOpenBatchNormActivFwdTrainSpatial.cl (hunk 3 above)
Comment (muralinr) [reply src/kernels/MIOpenBatchNormActivFwdTrainSpatial.cl]: Done!

File: src/kernels/MIOpenBatchNormBwdSpatial.cl (hunk 4 above)
Comment (muralinr) [reply src/kernels/MIOpenBatchNormBwdSpatial.cl]: Done!

File: src/kernels/MIOpenBatchNormFwdTrainSpatial.cl (hunk 5 above)
Comment (muralinr) [reply src/kernels/MIOpenBatchNormFwdTrainSpatial.cl]: Done!

File: src/kernels/batchnorm_functions.h (hunk 6 above)
Comment (muralinr) [reply src/kernels/batchnorm_functions.h]: Done!

File: src/kernels/batchnorm_functions.h (hunk 7 above)
Comment (muralinr) [reply src/kernels/batchnorm_functions.h:167]: I will keep it as it is now. thank you.

**Review by atamazov — DISMISSED** [review]
LGTM!

**Comment by atamazov** [comment]
@muralinr Please prepend `[GFX11]` to the title of this PR.

**Review by junliume — APPROVED** [review]

**Pull request merged**